documentation/source/newsfragments/2201.removal.rst

```diff
@@ -0,0 +1,5 @@
+Deprecated :class:`cocotb.hook` and :envvar:`COCOTB_HOOKS`.
+Write module-level Python code;
+create higher-priority tests using the ``stage`` option of :class:`cocotb.test`;
+or define your own function decorators;
+depending on how you were using hooks.
```

Comment:
```suggestion
See the documentation for :class:`cocotb.hook` for suggestions on alternatives.
```

Reply:
There is a danger here that once we remove them entirely, our release notes will contain dead links.

Perhaps the way to fix this is to turn on bad-references-as-errors in sphinx (`-n`), so that when we try to remove `hook` entirely, we're forced to think about this link.

Reply:
Once we release we could update the link to point to the 1.5 version notes by creating a new intersphinx mapping.